### PR TITLE
fix issue detecting go minimal version

### PR DESCRIPTION
**Pull request by sdminonne**
@mfojtik found an issue building github.com/open-cluster-management/api. This PR should fix it (as this code is vendored)
@pmorie  FYI

### CODE REVIEW

File: make/lib/golang.mk [hunk 1]
```diff
@@ -18,7 +18,10 @@ GOLINT ?=golint
 
 go_version :=$(shell $(GO) version | sed -E -e 's/.*go([0-9]+.[0-9]+.[0-9]+).*/\1/')
 GO_REQUIRED_MIN_VERSION ?=1.13.4
-ifneq "$(GO_REQUIRED_MIN_VERSION)" ""
+sorted_versions := $(sort $(GO_REQUIRED_MIN_VERSION) $(go_version))
+minVersion := $(word 1, $(sorted_versions))
+
+ifneq "$(GO_REQUIRED_MIN_VERSION)" "$(minVersion)"
```
Comment (tnozicka) [inline make/lib/golang.mk:24]: this was checking if GO_REQUIRED_MIN_VERSION is set

Comment (sdminonne) [reply make/lib/golang.mk:24]: Sure, I tried to compile  github.com/open-cluster-management/api and I was reported that my version `go version` which is 1.13.4 did not match the minimal required version which is 1.13.4. So I looked at  Makefile and I found GO_REQUIRED_MIN_VERSION was compared with `""`.

Comment (soltysh) [reply make/lib/golang.mk:24]: Based on the output I got from @sdminonne which is:
```
go version
go version go1.13.4 linux/amd64
:~/dev/src/github.com/open-cluster-management/api$ make
/bin/sh: 1: set: Illegal option -o pipefail
vendor/github.com/openshift/build-machinery-go/make/targets/golang/../../lib/golang.mk:22: *** `go` is required with minimal version "1.13.4", detected version "1.13.4". You can override this check by using `make GO_REQUIRED_MIN_VERSION:=`.  
```
I did a bit of googling and I found that `pipefail` is bashism (https://stackoverflow.com/questions/17757039/equivalent-of-pipefail-in-dash-shell) there's no easy alternative, but that would suggest why `GO_REQUIRED_MIN_VERSION` is empty. You're on ubuntu, and this is the root cause. Can you verify that element? Is it actually using the dash, which is linked to sh? Does switching to bash helps with this? Maybe we need better bash verification.

Comment (sdminonne) [reply make/lib/golang.mk:24]: @soltysh right. I'm on Ubuntu (don't ask me why :) ).
`$ ls -l /bin/sh
lrwxrwxrwx 1 root root 4 Apr  3  2019 /bin/sh -> dash
`
Fixing this I managed to compile, thanks for this, and sorry for the noise.

**Comment by sdminonne** [comment]
Using bash instead of `dash` solve the issue. Thanks @soltysh

**Pull request closed without merging**